libavcodec/qsv.c

```diff
@@ -390,11 +390,27 @@ int ff_qsv_init_internal_session(AVCodecContext *avctx, QSVSession *qs,
     const char *desc;
     int ret;
 
+#if QSV_VERSION_ATLEAST(1, 15)
+    mfxExtBuffer *ext_params[1];
+    mfxExtThreadsParam thread_param;
+#endif
+
 #if QSV_VERSION_ATLEAST(1, 16)
     init_par.GPUCopy        = gpu_copy;
 #endif
     init_par.Implementation = impl;
     init_par.Version        = ver;
+
+#if QSV_VERSION_ATLEAST(1, 15)
+    memset(&thread_param, 0, sizeof(thread_param));
+    thread_param.Header.BufferId = MFX_EXTBUFF_THREADS_PARAM;
+    thread_param.Header.BufferSz = sizeof(thread_param);
+    thread_param.NumThread       = FFMAX(2, avctx->thread_count);
+    ext_params[0]                = (mfxExtBuffer *)&thread_param;
+    init_par.ExtParam            = (mfxExtBuffer **)&ext_params;
+    init_par.NumExtParam         = 1;
+#endif
```

Comment:
duplicate code could we use one function for this?
qsv_init_mfx(init_par, session, maxthreads)

Reply:
libavcodec and libavutils are two different libraries. It seems the FFmpeg community doesn't prefer avpriv_ functions (which are used  internally across multiple libraries) and I prefer not to introduce new av_ functions

Reply:
thanks for the explain.
could we define static functions to reduce duplicate code in one file?

Reply:
No.  It will result in the 'duplicate definition' error when both libavxxx libraries are linked statically. In the past, we experienced such error and had to define functions with different names.